app/models/lightweight_activity.rb

```diff
@@ -154,7 +154,28 @@ def export
     self.plugins.each do |p|
       activity_json[:plugins] << p.export
     end
-    # TODO: add special code to push glossary plugin onto plugin list if glossary is set
+
+    # if the activity has a glossary model assiged to it, add the fake glossary plugin to the list of plugins
+    # replacing any existing glossary plugins and using the existing glossary plugin's approved script
+    if self.glossary_id && approved_glossary_script = ApprovedScript.find_by_label("glossary")
+      # remove any existing glossary script
+      activity_json[:plugins].delete_if { |plugin| plugin[:component_label] == "glossary" }
+
+      fake_glossary_plugin = {
+        id: 0,
+        description: nil,
+        author_data: JSON.generate({
+          version:"1.0",
+          glossaryResourceId: "this-is-a-fake-glossary-resource-id",
```

Comment:
If this is used in the student data saving or in some other way by the AP to manage plugins, this might need to be unique. If you know more about the requirements of it, adding a comment would be helpful.

Reply:
The `glossaryResourceId` is not used except by the glossary plugin to generate the s3Url.  This could have been left blank but we didn't.

Reply:
Can you add a comment in the code describing this? Given that you've built up a train a commits you could add it to another PR.